Mathlib/RingTheory/AdicCompletion/Basic.lean

```diff
@@ -518,10 +521,238 @@ lemma eval_lift_apply (f : ∀ (n : ℕ), M →ₗ[R] N ⧸ (I ^ n • ⊤ : Sub
     (n : ℕ) (x : M) : (lift I f h x).val n = f n x :=
   rfl
 
+section Bijective
+
+variable {I}
+
+theorem injective_of_iff : Function.Injective (of I M) ↔ IsHausdorff I M := by
+  constructor
+  · refine fun h ↦ ⟨fun x hx ↦ h ?_⟩
+    ext n
+    simpa [of, SModEq.zero] using hx n
+  · intro h
+    rw [← LinearMap.ker_eq_bot]
+    ext x
+    simp only [LinearMap.mem_ker, Submodule.mem_bot]
+    refine ⟨fun hx ↦ h.haus x fun n ↦ ?_, fun hx ↦ by simp [hx]⟩
+    rw [Subtype.ext_iff] at hx
+    simpa [SModEq.zero] using congrFun hx n
+
+variable (I M) in
+theorem injective_of [IsHausdorff I M] : Function.Injective (of I M) :=
```

Comment:
Can you follow the [naming convention](https://leanprover-community.github.io/contribute/naming.html#injectivity)? Also adding the `inj` version.